File: atcoder/denso2023_std/e.py
import copy

def dfs(l,ans):
    if l==[]: return ans
    max = l.pop(-1)
    for i in range(len(l)-1,-1,-1):
        if l[i]*2 < max: break
        ll = copy.copy(l)
        y = ll.pop(i)
        x = max-y
        if x in ll:
            ll.remove(x)
            ret = dfs(ll,ans+[[x,y,max]])
            if ret: return ret
    return False

n=int(input())
a=sorted(map(int,input().split()))

ans=dfs(a,[])
if ans:
    print("Yes")
    for a in ans:
        print(" ".join(map(str,a)))
else:
    print("No")

File: atcoder/denso2023_std/e.rb
def dfs(l,ans)
    return ans if l==[]
    z=l.pop
    (l.size-1).downto(0){|i|
        break if l[i]*2<z
        ll=l.dup
        y=ll.delete_at(i)
        if idx=ll.index(z-y)
            x=ll.delete_at(idx)
            if ret = dfs(ll,ans+[[x,y,z]])
                return ret
            end
        end
    }
    return false
end

n=gets.to_i
a=gets.split.map(&:to_i).sort

ans=dfs(a,[])
puts ans ? "Yes\n"+ans.map{|a|a*" "}*"\n" : "No"
